### PR TITLE
[dynamo] Graph break on ops having inplace_view tag

### PR DESCRIPTION
**Pull request by blzheng**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100787
* #100786

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_dynamo/variables/tensor.py [hunk 1]
```diff
@@ -184,11 +184,16 @@ def var_getattr(self, tx, name):
...
-            # Delay the graph break to the actual call of unsqueeze_/resize_/resize_as_
-            return variables.misc.DelayGraphBreakVariable()
+        if self.source is not None:
+            try:
+                fn = getattr(torch.ops.aten, name)
+                if torch.Tag.inplace_view in getattr(fn, fn.overloads()[0]).tags:
+                    # Delay the graph break to the actual call of unsqueeze_/resize_/resize_as_ etc.
+                    return variables.misc.DelayGraphBreakVariable()
+            except AttributeError:
+                pass
```
Comment (jgong5) [inline torch/_dynamo/variables/tensor.py:196]: can you do "if" check instead of try-catch?

File: test/dynamo/test_aot_autograd.py
Comment (jgong5) [inline test/dynamo/test_aot_autograd.py:1]: Is the change in this file related to the issue you are fixing?

Comment (blzheng) [reply test/dynamo/test_aot_autograd.py:1]: Yes. Inplace view (metadata mutation) on graph input trigger graph break, so ops satisfy such condition don't appear on the graph. As for the uts changed in this file, t_ is the op that will trigger graph break. As a result, I replaced t_ with trunc_ in some tests and removed the ut test_arg_metadata_mutation_on_input_causes_recompile.

File: torch/_dynamo/variables/tensor.py (hunk 1 above)
Comment (blzheng) [reply torch/_dynamo/variables/tensor.py]: Done

**Review by jgong5 — APPROVED** [review]

File: torch/_dynamo/variables/tensor.py [hunk 2]
```diff
@@ -184,11 +184,17 @@ def var_getattr(self, tx, name):
...
         # dynamo/aot/inductor, just fall back.
-        if name in ("resize_", "unsqueeze_", "resize_as_") and self.source is not None:
-            # Delay the graph break to the actual call of unsqueeze_/resize_/resize_as_
-            return variables.misc.DelayGraphBreakVariable()
+        if self.source is not None and hasattr(torch.ops.aten, name):
+            fn = getattr(torch.ops.aten, name)
+            if (
+                hasattr(fn, "overloads")
+                and hasattr(fn, fn.overloads()[0])
+                and torch.Tag.inplace_view in getattr(fn, fn.overloads()[0]).tags
```
Comment (eellison) [inline torch/_dynamo/variables/tensor.py:194]: we're only checking the first overload but that should be fine here..

**Review by eellison — APPROVED** [review]
nice!

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**